From da653280f73685dfa6bb1663ff60de8a264055de Mon Sep 17 00:00:00 2001
From: "Processe, Christian" <christian.processe@intel.com>
Date: Thu, 28 Apr 2016 15:39:22 +0100
Subject: [PATCH] Revert "FOR_UPSTREAM [VPG]: drm/i915: Fix audio checks during
 modeset"

This reverts commit 5793f81d8ff88ade0e22063a95a7efbf3006f61b
This reverts https://android.intel.com/#/c/498180

Change-Id: Ib3532f9314ffe8adad43188b8fd68a5bb4e0c5a6
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30675
Signed-off-by: geoffroy.weisenhorn <weisenhorn.geoffroy@intel.com>
Reviewed-on: https://android.intel.com:443/513802
---
 drivers/gpu/drm/i915/intel_display.c |  3 +--
 drivers/gpu/drm/i915/intel_dp.c      | 10 ++--------
 drivers/gpu/drm/i915/intel_hdmi.c    | 10 ++--------
 3 files changed, 5 insertions(+), 18 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 4d23545..e3c0aa2 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -12285,8 +12285,7 @@ intel_pipe_config_compare(struct drm_device *dev,
 	    IS_VALLEYVIEW(dev))
 		PIPE_CONF_CHECK_I(limited_color_range);
 
-	if (!IS_VALLEYVIEW(dev))
-		PIPE_CONF_CHECK_I(has_audio);
+	PIPE_CONF_CHECK_I(has_audio);
 
 	PIPE_CONF_CHECK_FLAGS(adjusted_mode.flags,
 			      DRM_MODE_FLAG_INTERLACE);
diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index a9b177d..a36808e 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -1247,15 +1247,9 @@ static void intel_dp_prepare(struct intel_encoder *encoder)
 	intel_dp->DP |= DP_VOLTAGE_0_4 | DP_PRE_EMPHASIS_0;
 	intel_dp->DP |= DP_PORT_WIDTH(intel_dp->lane_count);
 
-	/*
-	 * LP audio driver communicates through i915
-	 * where we enable audio output on port
-	 * hence not needed to be enabled during
-	 * modeset
-	 */
-	if (crtc->config.has_audio && !IS_VALLEYVIEW(dev)) {
+	if (crtc->config.has_audio) {
 		DRM_DEBUG_DRIVER("Enabling DP audio on pipe %c\n",
-				pipe_name(crtc->pipe));
+				 pipe_name(crtc->pipe));
 		intel_dp->DP |= DP_AUDIO_OUTPUT_ENABLE;
 		intel_write_eld(&encoder->base, adjusted_mode);
 	}
diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index a5533ec..d5ae6db 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -692,13 +692,7 @@ static void intel_hdmi_prepare(struct intel_encoder *encoder)
 	if (crtc->config.has_hdmi_sink)
 		hdmi_val |= HDMI_MODE_SELECT_HDMI;
 
-	/*
-	 * LP audio driver communicates through i915
-	 * where we enable audio output on port
-	 * hence not needed to be enabled during
-	 * modeset
-	 */
-	if (crtc->config.has_audio && !IS_VALLEYVIEW(dev)) {
+	if (crtc->config.has_audio) {
 		WARN_ON(!crtc->config.has_hdmi_sink);
 		DRM_DEBUG_DRIVER("Enabling HDMI audio on pipe %c\n",
 				 pipe_name(crtc->pipe));
@@ -793,7 +787,7 @@ static void intel_enable_hdmi(struct intel_encoder *encoder)
 	u32 temp;
 	u32 enable_bits = SDVO_ENABLE;
 
-	if (intel_crtc->config.has_audio && !IS_VALLEYVIEW(dev))
+	if (intel_crtc->config.has_audio)
 		enable_bits |= SDVO_AUDIO_ENABLE;
 
 	temp = I915_READ(intel_hdmi->hdmi_reg);
-- 
1.9.1

